RosaServer/api.cpp

```diff
@@ -1122,42 +1123,50 @@ std::string Vector::__tostring() const {
 }
 
 Vector Vector::__add(Vector* other) const {
+	if (!other) throw std::invalid_argument(missingArgument);
 	return {x + other->x, y + other->y, z + other->z};
 }
 
 Vector Vector::__sub(Vector* other) const {
+	if (!other) throw std::invalid_argument(missingArgument);
 	return {x - other->x, y - other->y, z - other->z};
 }
 
 Vector Vector::__mul(float scalar) const {
+	if (!scalar) throw std::invalid_argument(missingArgument);

```

Comment:
Not a pointer, shouldn't check this

Reply:
does sol make the base type params not null if not given

Reply:
There's no such thing as a null float